lib/removeUnusedElements.js

```diff
@@ -260,3 +324,81 @@ getListOfElementsIdsInUse.bufferView = function(gltf) {
 
     return usedBufferViewIds;
 };
+
+getListOfElementsIdsInUse.mesh = function(gltf) {
+    const usedMeshIds = {};
+    ForEach.mesh(gltf, (mesh, i) => {
+        if(!defined(mesh.primitives) || mesh.primitives.length === 0) {
+            usedMeshIds[i] = false;
+        }
+    });
+
+    ForEach.node(gltf, (node, i) => {
+        if(!defined(node.mesh)) {
+            return;
+        }
+        /* Mesh marked as empty in previous step? */
+        const meshIsEmpty = defined(usedMeshIds[node.mesh]);
+
+        if(!meshIsEmpty) {
+            usedMeshIds[node.mesh] = true;
+        }
+    });
+
+    return usedMeshIds;
+};
+
+/* Find parent node id of the given node, or -1 if no parent */
+function findParentNode(gltf, node) {
+    return gltf.nodes.findIndex(n => (n.children || []).includes(node));
+}
+
+/* Check if node is empty. It is considered empty if neither referencing
+ * mesh, camera, extensions and has no children */
+function nodeIsEmpty(gltf, node) {
+    if(defined(node.mesh) || defined(node.camera) || defined(node.skin)
+        || defined(node.weights) || defined(node.extras)
+        || (defined(node.extensions) && node.extensions.length !== 0)) {
+        return false;
+    }
+
+    /* Empty if no children or children are all empty nodes */
+    return !defined(node.children)
+        || node.children.filter(n => !nodeIsEmpty(gltf, gltf.nodes[n])).length === 0;
+}
+
+getListOfElementsIdsInUse.node = function(gltf) {
+    const usedNodeIds = {};
```

Comment:
Also handle:

* `skin.skeleton` and `skin.joints` referencing nodes
* [`KHR_techniques_webgl`](https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Khronos/KHR_techniques_webgl) referencing a node. Which would look like:

```
    ForEach.technique(gltf, function (technique) {
        ForEach.techniqueUniform(technique, function (uniform) {
            if (defined(uniform.node)) {
                usedNodeIds[uniform.node] = true;
            }
        });
    });
```

Reply:
Awesome! Was expecting something like this to pop up, thanks!